### PR TITLE
Migrates GravatarServiceRemote classes & models to WordPressKit

### LINKED ISSUE #7310: Migrate GravatarServiceRemote to WordPressKit framework
Refer to: #7301

Migrate the remote service class along with any related data model classes and unit tests.

### PR DESCRIPTION
**Pull request by bummytime**
Fixes #7310 

This PR:
* Migrates the `RemoteGravatarProfile ` model to the WPKit framework
* Migrates `GravatarServiceRemote` to the WPKit framework
* Migrates `NSMutableData+Helpers.swift` to the WPShared framework (used in main app and WPKit)
* Migrates `NSString+Helpers.h` & `NSString+Helpers.m` to the WPShared framework (used in main app and WPKit)
* Updates all imports for the files moved to the shared framework (and removed un-needed ones).

Ref #7301

**To test:**
1. Clean DerivedData.
2. Update CocoaPods repo with pod repo update.
3. Install the pods with pod install.
4. Run _all_ unit tests and make sure they are green.
5. Build & run the app:
a. Verify you can log into an .org, non-jetpack site and the Gravitar is properly fetched [via](https://github.com/wordpress-mobile/WordPress-iOS/blob/eb7fdae0f6a3fb9dd093ceb88933fa137547fced/WordPress/Classes/ViewRelated/NUX/LoginSelfHostedViewController.swift#L319)
b. Verify you can upload a new Gravitar image via the Me tab

Needs review, por favor @elibud. Thanks!! 😄

### CODE REVIEW

File: WordPressShared/WordPressShared.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -659,6 +671,7 @@
 			baseConfigurationReference = 3BEDE0D11F9FF330B15F3EBB /* Pods-WordPressShared.release-internal.xcconfig */;
 			buildSettings = {
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = "$(inherited)";
+				CLANG_WARN_DOCUMENTATION_COMMENTS = NO;
```
Comment (elibud) [inline WordPressShared/WordPressShared.xcodeproj/project.pbxproj]: I don't think we want this (and all of its repetitions below), we have it on YES for the other projects.

File: WordPressShared/WordPressShared/Core/Utility/NSString+Helpers.h [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (bummytime) [inline WordPressShared/WordPressShared/Core/Utility/NSString+Helpers.h]: Correct. I will remove it.

File: WordPressComStatsiOS/WordPressComStatsiOS.xcodeproj/xcshareddata/xcschemes/WordPressComStatsiOS.xcscheme [hunk 3]
```diff
@@ -52,6 +52,36 @@
                ReferencedContainer = "container:WordPressComStatsiOS.xcodeproj">
             </BuildableReference>
          </TestableReference>
+         <TestableReference
```
Comment (bummytime) [inline WordPressComStatsiOS/WordPressComStatsiOS.xcodeproj/xcshareddata/xcschemes/WordPressComStatsiOS.xcscheme]: Hmmm interesting. Let me investigate why this delta occurred. At one point today all of the tests in XCode were disabled so I re-enabled them. I wonder if that caused this mutation.

File: WordPressShared/WordPressShared.xcodeproj/project.pbxproj (hunk 1 above)
Comment (bummytime) [inline WordPressShared/WordPressShared.xcodeproj/project.pbxproj]: Fair enough.

**Comment by bummytime** [comment]
Ok @elibud ready for another look. Thank you!!

File: WordPressComStatsiOS/WordPressComStatsiOS.xcodeproj/xcshareddata/xcschemes/WordPressComStatsiOS.xcscheme (hunk 3 above)
Comment (elibud) [inline WordPressComStatsiOS/WordPressComStatsiOS.xcodeproj/xcshareddata/xcschemes/WordPressComStatsiOS.xcscheme]: Stats tests seem to be working correctly, I would still try to get rid of this so we don't make unnecessary changes.

**Comment by bummytime** [comment]
Thanks @elibud! I will revert the `<TestableReference>` addition.
